kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NMSettingsConverterTest.java

```diff
@@ -0,0 +1,603 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Eurotech and/or its affiliates and others
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *  Eurotech
+ *******************************************************************************/
+package org.eclipse.kura.nm.configuration;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.eclipse.kura.configuration.Password;
+import org.eclipse.kura.nm.NMDeviceType;
+import org.eclipse.kura.nm.NetworkProperties;
+import org.freedesktop.dbus.types.UInt32;
+import org.freedesktop.dbus.types.Variant;
+import org.freedesktop.networkmanager.settings.Connection;
+import org.junit.Test;
+
+public class NMSettingsConverterTest {
+
+	Map<String, Variant<?>> internalComparatorMap = new HashMap<>();
+	Map<String, Variant<?>> resultMap;
+
+	Map<String, Map<String, Variant<?>>> internalComparatorAllSettingsMap = new HashMap<>();
+	Map<String, Map<String, Variant<?>>> resultAllSettingsMap = new HashMap<>();
+
+	Map<String, Object> internetNetworkPropertiesInstanciationMap = new HashMap<>();
+
+	NetworkProperties networkProperties;
+
+	Boolean hasIllegalArgumentExceptionBeenThrown = false;
+	Boolean hasAGenericExecptionBeenThrown = false;
+
+	@Test
+	public void buildSettingsShouldThrowWhenGivenEmptyMap() {
+		givenNetworkPropsCreatedWithTheMap(this.internetNetworkPropertiesInstanciationMap);
+		whenBuildSettingsIsRunWithNetworkPropsAndIfaceString(this.networkProperties, Optional.empty(), "wlan0",
+				NMDeviceType.NM_DEVICE_TYPE_WIFI);
+		thenIllegalArgumentExceptionHasBeenThrown();
+	}
+
+	@Test
+	public void buildIpv4SettingsShouldThrowWhenGivenEmptyMap() {
+		givenNetworkPropsCreatedWithTheMap(this.internetNetworkPropertiesInstanciationMap);
+		whenBuildIpv4SettingsIsRunWithNetworkPropsAndIfaceString(this.networkProperties, "wlan0");
```

Comment:
This name (and others around) are unnecessarily complicated and hard to read. We already know what arguments we're passing, they're right there...

```suggestion
		whenBuildIpv4SettingsIsRunWith(this.networkProperties, "wlan0");
```